account_payment_mode_auto_reconcile/models/account_move.py

```diff
@@ -89,34 +97,49 @@ def _filter_payment_same_journal(self, credits_dict):
         )
         return [credit for credit in credits_dict if credit["id"] in lines.ids]
 
-    @api.multi
     def auto_unreconcile_credits(self):
         for invoice in self:
-            payments_info = json.loads(invoice.payments_widget or "{}")
+            payments_info = invoice.invoice_payments_widget
             for payment in payments_info.get("content", []):
-                aml = self.env["account.move.line"].browse(payment.get("payment_id"))
+                payment_aml = (
+                    self.env["account.payment"]
+                    .browse(payment.get("account_payment_id"))
+                    .line_ids
+                )
+
+                aml = payment_aml.filtered(lambda l: l.matched_debit_ids)
                 for apr in aml.matched_debit_ids:
                     if apr.amount != payment.get("amount"):
                         continue
                     if (
                         apr.payment_mode_auto_reconcile
-                        and apr.debit_move_id.invoice_id == invoice
+                        and apr.debit_move_id.move_id == invoice
                     ):
-                        aml.with_context(invoice_id=invoice.id).remove_move_reconcile()
+                        aml.remove_move_reconcile()
 
     @api.depends(
-        "type", "payment_mode_id", "payment_move_line_ids", "state", "has_outstanding"
+        "move_type", "payment_mode_id", "payment_id", "state", "invoice_has_outstanding"
     )
     def _compute_payment_mode_warning(self):
         # TODO Improve me but watch out
         for invoice in self:
-            if invoice.type != "out_invoice" or invoice.state == "paid":
+            existed_reconciliations = any(
+                [
+                    invoice.line_ids.mapped("matched_credit_ids"),
+                    invoice.line_ids.mapped("matched_debit_ids"),
+                ]
+            )
+            if (
+                invoice.move_type != "out_invoice"
+                or invoice.state == "posted"
```

Comment:
it should be `or ( invoice.state == "posted" and invoice.payment_state != "paid")`